src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/FromKeyedServicesAttribute.cs

```diff
@@ -8,18 +8,40 @@ namespace Microsoft.Extensions.DependencyInjection
     /// <summary>
     /// Indicates that the parameter should be bound using the keyed service registered with the specified key.
     /// </summary>
+    /// <seealso cref="ServiceKeyAttribute"/>
     [AttributeUsage(AttributeTargets.Parameter)]
     public class FromKeyedServicesAttribute : Attribute
     {
         /// <summary>
         /// Creates a new <see cref="FromKeyedServicesAttribute"/> instance.
         /// </summary>
         /// <param name="key">The key of the keyed service to bind to.</param>
-        public FromKeyedServicesAttribute(object key) => Key = key;
+        public FromKeyedServicesAttribute(object? key)
+        {
+            Key = key;
+            LookupMode = key == null ? ServiceKeyLookupMode.NullKey : ServiceKeyLookupMode.ExplicitKey;
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="FromKeyedServicesAttribute"/> instance with <see cref="LookupMode"/> set to <see cref="ServiceKeyLookupMode.InheritKey"/>.
+        /// </summary>
+        public FromKeyedServicesAttribute()
+        {
+            Key = null;
+            LookupMode = ServiceKeyLookupMode.InheritKey;
+        }
 
         /// <summary>
         /// The key of the keyed service to bind to.
         /// </summary>
-        public object Key { get; }
+        /// <remarks>A <see langword="null"/> value indicates there is not a key and just the parameter type is used to resolve the service.
+        /// This is useful for DI implementations that require an explict way to declare that the parameter should be resolved for unkeyed services.
```

Comment:
Is this still fully correct now that there's also InheritKey? This can be null for both NullKey and InheritKey.